packages/lib/CloseCom.ts

```diff
@@ -10,66 +25,245 @@ export type CloseComStatus = {
   label: string;
 };
 
+export type CloseComCustomActivityTypeCreate = {
+  name: string;
+  description: string;
+};
+
+export type CloseComContactSearch = {
+  data: {
+    __object_type: "contact";
+    emails: {
+      email: string;
+      type: string;
+    }[];
+    id: string;
+    lead_id: string;
+    name: string;
+  }[];
+  cursor: null;
+};
+
+export type CloseComCustomActivityTypeGet = {
+  data: {
+    api_create_only: boolean;
+    created_by: string;
+    date_created: string;
+    date_updated: string;
+    description: string;
+    editable_with_roles: string[];
+    fields: CloseComCustomActivityFieldGet["data"][number][];
+    id: string;
+    name: string;
+    organization_id: string;
+    updated_by: string;
+  }[];
+  cursor: null;
+};
+
+export type CloseComCustomActivityFieldCreate = {
+  custom_activity_type_id: string;
+  name: string;
+  type: string;
+  required: boolean;
+  accepts_multiple_values: boolean;
+  editable_with_roles: string[];
+};
+
+export type CloseComCustomActivityFieldGet = {
+  data: {
+    custom_activity_type_id: string;
+    id: string;
+    name: string;
+    description: string;
+    type: string;
+    required: boolean;
+    accepts_multiple_values: boolean;
+    editable_with_roles: string[];
+    created_by: string;
+    updated_by: string;
+    date_created: string;
+    date_updated: string;
+    organization_id: string;
+  }[];
+};
+
+export type CloseComCustomActivityCreate = {
+  custom_activity_type_id: string;
+  lead_id: string;
+  [key: CloseComCustomActivityCustomField<string>]: string;
+};
+
+export type typeCloseComCustomActivityGet = {
+  organization_id: string;
+  contact_id: any;
+  date_updated: string;
+  user_name: string;
+  created_by_name: "Bruce Wayne";
+  id: string;
+  created_by: string;
+  status: string;
+  user_id: string;
+  users: any[];
+  lead_id: string;
+  _type: string;
+  updated_by: string;
+  custom_activity_type_id: string;
+  date_created: string;
+  updated_by_name: string;
+  [key: CloseComCustomActivityCustomField<string>]: string;
+};
+
+type CloseComCustomActivityCustomField<T extends string> = `custom.${T}`;
+
+const environmentApiKey = process.env.CLOSECOM_API_KEY || "";
```

Comment:
Can we fetch this from the DB? you can [follow this style](https://github.com/calcom/cal.com/blob/feat/closecom-app/packages/app-store/office365calendar/lib/CalendarService.ts#L243-L253) to achieve that.

Reply:
We actually have two different ways to instance CloseCom. With a key that comes from the user, to manipulate their instance of CloseCom or having one for Cal.com for which that will come from an environment variable.

Reply:
Let's sync about this @leog

Reply:
Sure. PTAL at https://github.com/calcom/cal.com/pull/3814 to know more to what I meant about instancing CloseCom for Cal.com purposes.